src/commands/commandgraphicguess.cpp

```diff
@@ -0,0 +1,93 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "commandgraphicguess.h"
+#include "commandlineparser.h"
+#include "leakdetector.h"
+#include "logger.h"
+
+#include <QDir>
+#include <QFile>
+#include <QGuiApplication>
+#include <QQmlApplicationEngine>
+#include <QStandardPaths>
+#include <QTextStream>
+
+namespace {
+Logger logger(LOG_MAIN, "CommandGraphicGuess");
+
+uint32_t s_vertexShaderErrorCount = 0;
+void messageQTHandler(QtMsgType, const QMessageLogContext&,
+                      const QString& message) {
+  if (message.contains("0x80070057")) {
+    ++s_vertexShaderErrorCount;
+  }
+}
+
+}  // namespace
+
+CommandGraphicGuess::CommandGraphicGuess(QObject* parent)
+    : Command(parent, "gg", "Guess the graphic configuration") {
+  MVPN_COUNT_CTOR(CommandGraphicGuess);
+}
+
+CommandGraphicGuess::~CommandGraphicGuess() {
+  MVPN_COUNT_DTOR(CommandGraphicGuess);
+}
+
+int CommandGraphicGuess::run(QStringList& tokens) {
+  Q_ASSERT(!tokens.isEmpty());
+
+  // Let's take the location before creating QGuiApplication.
+  QString location =
```

Comment:
In terms of standards paths, this will result in `%AppData%\Roaming` and the output of this test is one of the few artifacts that is more appropriate to go into `%AppData%\Local`, which is for files that should *not* be migrated across machines.

Reply:
I would like to keep this new file next to the setting file. In this way we do not spread config files in random locations.